pandas/io/parsers.py

```diff
@@ -1714,6 +1714,7 @@ def _set_noconvert_columns(self):
             # A set of integers will be converted to a list in
             # the correct order every single time.
             usecols = list(self.usecols)
+            usecols.sort()
```

Comment:
self.usecols is a ``set`` of the original usecols. On PyPy ``set``s are not sorted so "every single time" requires a sort. 
```
>>>> list(set([3, 0, 2]))
[3, 0, 2]
```
Running tests with this line replaces by ``usecols.reverse() will show the dependency on the list being sorted:

Reply:
no averse to this, but why does this matter?

Reply:
can you add a test that exercises the sortedness?

Reply:
there are tests, but they all pass on CPython since sets of ints are always sorted, ie pandas/tests/io/parsers/usecol.py, lines 199, 270, 296 which all failed on PyPy before this change. Not sure how I could construct a failing test for CPython